tools/reccomp/reccomp.py

```diff
@@ -1,66 +1,44 @@
 #!/usr/bin/env python3
 
+import argparse
 from capstone import *
 import difflib
 import struct
 import subprocess
 import os
 import sys
 
-def print_usage():
-  print('Usage: %s [options] <original-binary> <recompiled-binary> <recompiled-pdb> <decomp-dir>\n' % sys.argv[0])
-  print('\t-v, --verbose <offset>\t\t\tPrint assembly diff for specific function (original file\'s offset)')
-  print('\t-h, --html <output-file>\t\t\tGenerate searchable HTML summary of status and diffs')
-  sys.exit(1)
+parser = argparse.ArgumentParser(allow_abbrev=False)
+parser.add_argument('original', metavar='original-binary', help='An original LEGO\N{Copyright Sign} binary')
+parser.add_argument('recompiled', metavar='recompiled-binary', help='The recompiled binary')
+parser.add_argument('pdb', metavar='recompiled-pdb', help='The PDB of the recompiled binary')
+parser.add_argument('decomp_dir', metavar='decomp-dir-pdb', help='The recompiled source tree')
+parser.add_argument('--verbose', '-v', metavar='offset', help='Print assembly diff for specific function (original file\'s offset)')
+parser.add_argument('--html', '-H', metavar='output-file', help='Generate searchable HTML summary of status and diffs')
 
-positional_args = []
-verbose = None
-skip = False
-html = None
-
-for i, arg in enumerate(sys.argv):
-  if skip:
-    skip = False
-    continue
-
-  if arg.startswith('-'):
-    # A flag rather than a positional arg
-    flag = arg[1:]
-
-    if flag == 'v' or flag == '-verbose':
-      verbose = int(sys.argv[i + 1], 16)
-      skip = True
-    elif flag == 'h' or flag == '-html':
-      html = sys.argv[i + 1]
-      skip = True
-    else:
-      print('Unknown flag: %s' % arg)
-      print_usage()
-  else:
-    positional_args.append(arg)
+args = parser.parse_args()
 
-if len(positional_args) != 5:
-  print_usage()
+skip = args.verbose or args.html
```

Comment:
The `skip` variable was only used for the arg parsing loop and is no longer used without it.

Reply:
No more skip